database/src/main/kotlin/no/nav/su/se/bakover/database/sak/FerdigeBehandlingerRepo.kt

```diff
@@ -0,0 +1,206 @@
+            behandlingStartet = tidspunkt("opprettet"),
+        )
+    }
+
+    private fun Row.hentStatus(
+        behandlingsTypeDB: BehandlingsTypeDB,
+        erBehandlingAvsluttet: Boolean,
+    ): Behandlingsoversikt.Behandlingsstatus {
+        if (erBehandlingAvsluttet) {
+            return Behandlingsoversikt.Behandlingsstatus.AVSLUTTET
+        }
+        return when (behandlingsTypeDB) {
+            BehandlingsTypeDB.SØKNADSBEHANDLING -> BehandlingsStatus.valueOf(string("status"))
+                .tilBehandlingsoversiktStatus()
+            BehandlingsTypeDB.REVURDERING -> RevurderingsType.valueOf(string("status"))
+                .tilBehandlingsoversiktStatus()
+            BehandlingsTypeDB.KLAGE -> KlagePostgresRepo.Tilstand.fromString(string("status"))
+                .tilBehandlingsoversiktStatus()
+        }
+    }
+
+    private fun BehandlingsStatus.tilBehandlingsoversiktStatus(): Behandlingsoversikt.Behandlingsstatus {
+        return when (this) {
+            BehandlingsStatus.OPPRETTET,
+            BehandlingsStatus.VILKÅRSVURDERT_INNVILGET,
+            BehandlingsStatus.VILKÅRSVURDERT_AVSLAG,
+            BehandlingsStatus.BEREGNET_INNVILGET,
+            BehandlingsStatus.BEREGNET_AVSLAG,
+            BehandlingsStatus.SIMULERT,
+            BehandlingsStatus.TIL_ATTESTERING_INNVILGET,
+            BehandlingsStatus.TIL_ATTESTERING_AVSLAG,
+            BehandlingsStatus.UNDERKJENT_INNVILGET,
+            BehandlingsStatus.UNDERKJENT_AVSLAG,
+            -> throw IllegalStateException("Behandlinger som ikke er iverksatt, avslått, eller avsluttet er ikke en ferdig behandling")
+
+            BehandlingsStatus.IVERKSATT_INNVILGET -> Behandlingsoversikt.Behandlingsstatus.INNVILGET
+
+            BehandlingsStatus.IVERKSATT_AVSLAG -> Behandlingsoversikt.Behandlingsstatus.AVSLAG
+        }
+    }
+
+    private fun RevurderingsType.tilBehandlingsoversiktStatus(): Behandlingsoversikt.Behandlingsstatus {
+        return when (this) {
+            RevurderingsType.OPPRETTET,
+            RevurderingsType.BEREGNET_INNVILGET,
+            RevurderingsType.BEREGNET_OPPHØRT,
+            RevurderingsType.BEREGNET_INGEN_ENDRING,
+            RevurderingsType.SIMULERT_INNVILGET,
+            RevurderingsType.SIMULERT_OPPHØRT,
+            RevurderingsType.SIMULERT_STANS,
+            RevurderingsType.SIMULERT_GJENOPPTAK,
+            RevurderingsType.TIL_ATTESTERING_INNVILGET,
+            RevurderingsType.TIL_ATTESTERING_OPPHØRT,
+            RevurderingsType.TIL_ATTESTERING_INGEN_ENDRING,
+            RevurderingsType.UNDERKJENT_INNVILGET,
+            RevurderingsType.UNDERKJENT_OPPHØRT,
+            RevurderingsType.UNDERKJENT_INGEN_ENDRING,
+            -> throw IllegalStateException("Behandlinger som ikke er iverksatt, avslått, eller avsluttet er ikke en ferdig behandling")
+
+            RevurderingsType.IVERKSATT_STANS -> Behandlingsoversikt.Behandlingsstatus.AVSLAG
```

Comment:
Dette er vel en litt rar mapping. Burde kanskje ha noe custom?

Reply:
Tok det opp med Loritha. vi var begge litt usikker på hva det egentlig skulle være. Med tanke på at det ikke er så krise, så lot vi det være avslag for stans, og heller endre det senere. Det kommer sikkert opp igjen

Reply:
Du kan evt. sjekke med John Are her @RamziAbuQassim . Mulig at de ikke ønsker at stans skal dukke opp i denne lista.